src/commandDetails/coin/leaderboard.ts

```diff
@@ -47,9 +48,16 @@ const getCurrentCoinLeaderboardEmbed = async (
       .join('\\_')
       .split('`')
       .join('\\`');
-    const userCoinEntryText = `${leaderboardArray.length + 1}. ${cleanUserTag} - ${
+    const userCoinEntryText = `${rank}. ${cleanUserTag} - ${
       userCoinEntry.balance
     } ${getCoinEmoji()}`;
+    if (userCoinEntry.user_id === currentUserId) {
+      currentPosition = rank;
+    }
+    if (previousBalance !== userCoinEntry.balance) {
+      previousBalance = userCoinEntry.balance;
+      rank = rank + 1;
```

Comment:
By this logic, the leaderboard would look something like the following:
```
1. User1 - 400
2. User2 - 300
3. User3 - 300
3. User4 - 300
3. User5 - 200
4. User6 - 100
```

Reply:
Updated the logic to display this correctly